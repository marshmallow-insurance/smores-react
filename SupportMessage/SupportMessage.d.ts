import { FC, MouseEventHandler, ReactElement } from 'react';
import { MarginProps } from '../utils/space';
type SupportMessageType = 'info' | 'fallbackStyle' | 'alert' | 'warning' | 'success';
export type SupportMessageProps = {
    className?: string;
    description: string | ReactElement;
    onClick?: MouseEventHandler;
    type: SupportMessageType;
    title?: string;
} & MarginProps;
export declare const SupportMessage: FC<SupportMessageProps>;
export {};
