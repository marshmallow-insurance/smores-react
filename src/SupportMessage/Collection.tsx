import React, { FC } from 'react'
import styled from 'styled-components'

import { Text } from '../Text'
import { Box } from '../Box'
import { SupportMessage, SupportMessageProps } from './SupportMessage'

const variationList: SupportMessageProps[] = [
  {
    type: 'info',
    title: 'Additional information',
    description: 'Information needs to be communicated to the user.',
  },
  {
    type: 'fallback',
    title: 'Additional information',
    description: 'Information needs to be communicated to the user.',
  },
  {
    type: 'alert',
    title: 'Potential issue',
    description: 'The user should be aware and possibly exercise caution.',
  },
  {
    type: 'warning',
    title: 'Error encountered',
    description: "An error has occurred, here's feedback on how to proceed.",
  },
  {
    type: 'success',
    title: 'Error encountered',
    description: "An error has occurred, here's feedback on how to proceed.",
  },
]

export const CollectionPage: FC = () => {
  return (
    <Box flex direction="column">
      {variationList.map(({ type, title, description }) => {
        return (
          <Box key={type} flex direction="column" mb="32px">
            <Title tag="span" typo="header-medium">
              Type: {type}
            </Title>
            <Box flex>
              <Label tag="span" typo="header-small">
                Without title
              </Label>
              <SupportMessage type={type} description={description} />
            </Box>
            <Box flex mt="8px">
              <Label tag="span" typo="header-small">
                With title
              </Label>
              <SupportMessage
                type={type}
                title={title}
                description={description}
              />
            </Box>
            <Box flex mt="8px">
              <Label tag="span" typo="header-small">
                Clickable
              </Label>
              <SupportMessage
                type={type}
                title={title}
                description={description}
                onClick={() => alert('Clicked!')}
              />
            </Box>
          </Box>
        )
      })}
    </Box>
  )
}

const LABEL_WIDTH = '140px'

const Title = styled(Text)`
  margin-bottom: 12px;
  font-size: 18px;
  line-height: 20px;
  margin-left: ${LABEL_WIDTH};
`

const Label = styled(Text)`
  margin-bottom: 4px;
  font-size: 14px;
  width: ${LABEL_WIDTH};
  flex-shrink: 0;
`
