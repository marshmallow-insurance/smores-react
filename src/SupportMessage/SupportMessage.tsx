import React, { FC } from 'react'
import styled, { css } from 'styled-components'
import { lighten } from 'polished'

import { Box } from '../Box'
import { Icon } from '../Icon'
import { theme } from '../theme'

export type SupportMessageProps = {
  className?: string
  children: string
  type: 'info' | 'alert' | 'warning'
  title?: string
  hasBackground?: boolean
  hasBorder?: boolean
  borderColor?: string
}

export const SupportMessage: FC<SupportMessageProps> = ({
  className,
  children,
  type = 'info',
  hasBorder = false,
  borderColor,
  title,
  hasBackground = true,
}) => (
  <Wrapper
    className={className}
    type={type}
    hasBorder={hasBorder}
    borderColor={borderColor}
    hasBackground={hasBackground}
  >
    <IconWrapper>
      <Icon
        size={24}
        render={type}
        color={
          type === 'warning'
            ? 'error'
            : type === 'alert'
            ? 'agentWarning'
            : 'secondary'
        }
      />
    </IconWrapper>
    <Box flex direction="column">
      {title && <Title>{title}</Title>}
      {children}
    </Box>
  </Wrapper>
)

interface IWrapper {
  type: SupportMessageProps['type']
  hasBorder?: boolean
  borderColor?: string
  hasBackground?: boolean
}

const IconWrapper = styled(Box)`
  align-self: center;
`

const Wrapper = styled.div<IWrapper>(
  ({ type, hasBorder, borderColor, hasBackground }) => css`
    align-items: center;
    ${hasBackground &&
    `background-color: ${
      type === 'info'
        ? theme.colors.background
        : type === 'alert'
        ? theme.colors.bgSecondary
        : lighten(0.45, theme.colors.error)
    };
    };`}
    box-sizing: border-box;
    ${hasBorder && `border: 1px solid ${borderColor};`}
    border-radius: 8px;
    margin-bottom: 24px;
    padding: 16px;
    display: flex;
    font-family: ${theme.font.system};
    color: ${theme.colors.secondary};
    font-size: 12px;

    span {
      margin: 0 16px 0 0;
    }
  `,
)

const Title = styled.h3`
  font-size: 16px;
  font-weight: ${theme.font.weight.medium};
  color: ${theme.colors.secondary};
  line-height: 20.8px;
  margin-bottom: 4px;
`
