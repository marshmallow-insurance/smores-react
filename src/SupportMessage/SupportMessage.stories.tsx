import React from 'react'
import { SupportMessage, SupportMessageProps } from './SupportMessage'
import { Link } from '../Link'
import { CollectionPage } from './Collection'

export default {
  title: 'SupportMessage',
  component: SupportMessage,
}

const Template = (props: SupportMessageProps) => (
  <SupportMessage {...props}>Support message for customer</SupportMessage>
)

export const Default = Template.bind({})

const supportMessageArgs: SupportMessageProps = {
  type: 'info',
  title: undefined,
  description: 'Some description text',
}

Default.args = supportMessageArgs

export const WithCustomDescription = Template.bind({})

WithCustomDescription.args = {
  type: 'info',
  title: 'A SupportMessage using the Link component',
  description: (
    <>
      Some text rendered using a <Link href={''}>Link</Link>
    </>
  ),
}

export const Clickable = Template.bind({})

Clickable.args = {
  type: 'info',
  title: 'An interactive SupportMessage',
  description: 'Click me!',
  onClick: () => alert('Clicked!'),
}

export const Collection = CollectionPage.bind({})
