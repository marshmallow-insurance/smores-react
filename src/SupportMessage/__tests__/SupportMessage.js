import React from 'react'
import { render } from '@testing-library/react'
import '@testing-library/jest-dom/extend-expect'
import 'jest-styled-components'

import { SupportMessage } from '../SupportMessage'

describe('SupportMessage component', () => {
  it('default renders correctly', () => {
    const { container } = render(
      <SupportMessage type="info">
        Support info message for customer
      </SupportMessage>,
    )

    expect(container).toMatchSnapshot()
  })

  it('type info with general title renders correctly', () => {
    const { container } = render(
      <SupportMessage type="info" title="General info title">
        Support info message for customer
      </SupportMessage>,
    )

    expect(container).toMatchSnapshot()
  })

  it('type alert renders correctly', () => {
    const { container } = render(
      <SupportMessage type="alert">
        Support alert message for customer
      </SupportMessage>,
    )

    expect(container).toMatchSnapshot()
  })

  it('type warning renders correctly', () => {
    const { container } = render(
      <SupportMessage type="warning">
        Support alert message for customer
      </SupportMessage>,
    )

    expect(container).toMatchSnapshot()
  })
})
