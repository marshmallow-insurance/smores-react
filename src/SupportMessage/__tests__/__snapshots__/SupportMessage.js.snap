// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SupportMessage component default renders correctly 1`] = `
.c3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  width: 24px;
  height: 24px;
  -webkit-transform: rotate(0deg);
  -ms-transform: rotate(0deg);
  transform: rotate(0deg);
  margin-top: 0;
  margin-right: 0;
  margin-bottom: 0;
  margin-left: 0;
}

.c1 {
  -webkit-align-self: center;
  -ms-flex-item-align: center;
  align-self: center;
}

.c0 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background-color: #F4F4F4;
  box-sizing: border-box;
  border-radius: 8px;
  margin-bottom: 24px;
  padding: 16px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  font-family: -apple-system,BlinkMacSystemFont,"Segoe UI",Roboto,Helvetica,Arial,sans-serif,"Apple Color Emoji","Segoe UI Emoji","Segoe UI Symbol";
  color: #112035;
  font-size: 12px;
}

.c0 span {
  margin: 0 16px 0 0;
}

<div>
  <div
    class="c0"
    type="info"
  >
    <div
      class="c1"
    >
      <span
        class="c2"
        color="secondary"
        rotate="0"
        size="24"
      >
        <svg
          fill="none"
          height="100%"
          viewBox="0 0 24 24"
          width="100%"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            clip-rule="evenodd"
            d="M12 0C5.383 0 0 5.383 0 12s5.383 12 12 12 12-5.383 12-12S18.617 0 12 0zm-1 6.99a1 1 0 112 0c0 .553-.447 1.005-1 1.005-.553 0-1-.442-1-.995v-.01zM11 11a1 1 0 112 0v7a1 1 0 11-2 0v-7zm-9 1c0 5.514 4.486 10 10 10s10-4.486 10-10S17.514 2 12 2 2 6.486 2 12z"
            fill="#112035"
            fill-rule="evenodd"
          />
        </svg>
      </span>
    </div>
    <div
      class="c3"
      direction="column"
    >
      Support info message for customer
    </div>
  </div>
</div>
`;

exports[`SupportMessage component type alert renders correctly 1`] = `
.c3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  width: 24px;
  height: 24px;
  -webkit-transform: rotate(0deg);
  -ms-transform: rotate(0deg);
  transform: rotate(0deg);
  margin-top: 0;
  margin-right: 0;
  margin-bottom: 0;
  margin-left: 0;
}

.c1 {
  -webkit-align-self: center;
  -ms-flex-item-align: center;
  align-self: center;
}

.c0 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background-color: #FAF3EE;
  box-sizing: border-box;
  border-radius: 8px;
  margin-bottom: 24px;
  padding: 16px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  font-family: -apple-system,BlinkMacSystemFont,"Segoe UI",Roboto,Helvetica,Arial,sans-serif,"Apple Color Emoji","Segoe UI Emoji","Segoe UI Symbol";
  color: #112035;
  font-size: 12px;
}

.c0 span {
  margin: 0 16px 0 0;
}

<div>
  <div
    class="c0"
    type="alert"
  >
    <div
      class="c1"
    >
      <span
        class="c2"
        color="agentWarning"
        rotate="0"
        size="24"
      >
        <svg
          fill="none"
          height="100%"
          viewBox="0 0 24 24"
          width="100%"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            clip-rule="evenodd"
            d="M21 0H3C1.346 0 0 1.346 0 3v13c0 1.654 1.346 3 3 3h4.52l3.699 4.625a1 1 0 001.562 0L16.48 19H21c1.654 0 3-1.346 3-3V3c0-1.654-1.346-3-3-3zM11 6a1 1 0 112 0v4a1 1 0 11-2 0V6zm0 6.99a1 1 0 112 0c0 .552-.447 1.005-1 1.005-.553 0-1-.443-1-.995v-.01zM21 17c.552 0 1-.449 1-1V3c0-.551-.448-1-1-1H3c-.552 0-1 .449-1 1v13c0 .551.448 1 1 1h5a1 1 0 01.781.375L12 21.399l3.219-4.024A1 1 0 0116 17h5z"
            fill="#FF8B00"
            fill-rule="evenodd"
          />
        </svg>
      </span>
    </div>
    <div
      class="c3"
      direction="column"
    >
      Support alert message for customer
    </div>
  </div>
</div>
`;

exports[`SupportMessage component type info with general title renders correctly 1`] = `
.c3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  width: 24px;
  height: 24px;
  -webkit-transform: rotate(0deg);
  -ms-transform: rotate(0deg);
  transform: rotate(0deg);
  margin-top: 0;
  margin-right: 0;
  margin-bottom: 0;
  margin-left: 0;
}

.c1 {
  -webkit-align-self: center;
  -ms-flex-item-align: center;
  align-self: center;
}

.c0 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background-color: #F4F4F4;
  box-sizing: border-box;
  border-radius: 8px;
  margin-bottom: 24px;
  padding: 16px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  font-family: -apple-system,BlinkMacSystemFont,"Segoe UI",Roboto,Helvetica,Arial,sans-serif,"Apple Color Emoji","Segoe UI Emoji","Segoe UI Symbol";
  color: #112035;
  font-size: 12px;
}

.c0 span {
  margin: 0 16px 0 0;
}

.c4 {
  font-size: 16px;
  font-weight: 500;
  color: #112035;
  line-height: 20.8px;
  margin-bottom: 4px;
}

<div>
  <div
    class="c0"
    type="info"
  >
    <div
      class="c1"
    >
      <span
        class="c2"
        color="secondary"
        rotate="0"
        size="24"
      >
        <svg
          fill="none"
          height="100%"
          viewBox="0 0 24 24"
          width="100%"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            clip-rule="evenodd"
            d="M12 0C5.383 0 0 5.383 0 12s5.383 12 12 12 12-5.383 12-12S18.617 0 12 0zm-1 6.99a1 1 0 112 0c0 .553-.447 1.005-1 1.005-.553 0-1-.442-1-.995v-.01zM11 11a1 1 0 112 0v7a1 1 0 11-2 0v-7zm-9 1c0 5.514 4.486 10 10 10s10-4.486 10-10S17.514 2 12 2 2 6.486 2 12z"
            fill="#112035"
            fill-rule="evenodd"
          />
        </svg>
      </span>
    </div>
    <div
      class="c3"
      direction="column"
    >
      <h3
        class="c4"
      >
        General info title
      </h3>
      Support info message for customer
    </div>
  </div>
</div>
`;

exports[`SupportMessage component type warning renders correctly 1`] = `
.c3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  width: 24px;
  height: 24px;
  -webkit-transform: rotate(0deg);
  -ms-transform: rotate(0deg);
  transform: rotate(0deg);
  margin-top: 0;
  margin-right: 0;
  margin-bottom: 0;
  margin-left: 0;
}

.c1 {
  -webkit-align-self: center;
  -ms-flex-item-align: center;
  align-self: center;
}

.c0 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background-color: #fceeee;
  box-sizing: border-box;
  border-radius: 8px;
  margin-bottom: 24px;
  padding: 16px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  font-family: -apple-system,BlinkMacSystemFont,"Segoe UI",Roboto,Helvetica,Arial,sans-serif,"Apple Color Emoji","Segoe UI Emoji","Segoe UI Symbol";
  color: #112035;
  font-size: 12px;
}

.c0 span {
  margin: 0 16px 0 0;
}

<div>
  <div
    class="c0"
    type="warning"
  >
    <div
      class="c1"
    >
      <span
        class="c2"
        color="error"
        rotate="0"
        size="24"
      >
        <svg
          fill="none"
          height="100%"
          viewBox="0 0 25 24"
          width="100%"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M12.001 6a1 1 0 00-1 1v6a1 1 0 102 0V7a1 1 0 00-1-1zm0 10.005a1.003 1.003 0 000 2.005 1 1 0 001-1V17a.995.995 0 00-1-.995zm11.42-.503l-7.64-13.316A4.31 4.31 0 0012.001 0a4.31 4.31 0 00-3.78 2.186L.581 15.502a4.288 4.288 0 00.01 4.327A4.313 4.313 0 004.36 22h15.28a4.313 4.313 0 003.771-2.171 4.288 4.288 0 00.01-4.327zm-1.739 3.324A2.333 2.333 0 0119.642 20H4.361a2.331 2.331 0 01-2.042-1.174 2.306 2.306 0 01-.005-2.328l7.64-13.316A2.332 2.332 0 0112.001 2c.856 0 1.621.441 2.046 1.182l7.64 13.316c.418.73.417 1.6-.005 2.328z"
            fill="#D82C2C"
          />
        </svg>
      </span>
    </div>
    <div
      class="c3"
      direction="column"
    >
      Support alert message for customer
    </div>
  </div>
</div>
`;
